docs/DesignDocs/NotesOnAttestationAPI.md

```diff
@@ -0,0 +1,160 @@
+the evidence formats according to their defined security policies.
+  - The OE SDK attestation API enables the attesters and verifiers to discover
+  and register supported plugins that are built, configured, and working
+  properly. For a specific attester or verifier instance, the supported list
+  of evidence is not required to perfectly match the list defined in the
+  security policies.
+
+# Options for Enclave-side Plugin Library Initialization
+
+As described in the design document
+[Attestation: OE SDK Integration with Intel® SGX SDK quote-ex Library for Generation of Evidence in New Formats](https://github.com/openenclave/openenclave/blob/master/docs/DesignDocs/SGX_QuoteEx_Integration.md),
+enclave-side plugin library initialization of attester plugins can be
+triggered by application enclave call of the OE SDK API
+`oe_get_attester_plugins()`.
+During initialization, the plugin library enumerates supported evidence
+formats, and returns a list of supported attester plugins.
+
+In general, there are several options for initialization of enclave-side
+plugin libraries while keeping application enclaves TEE agnostic.
+
+In this section, we focus on the discussion of the options for initialization
+of attester plugins. Similar options are available for initialization of
+verifier plugins.
+
+1. Every enclave-side plugin library implements the same OE SDK API
+`oe_get_attester_plugins()`.
+
+    With this option, a build of an application enclave project can only link
+    with a single attestation plugin library, to avoid function name collision.
+
+2. Compile-time macro to convert from TEE-agnostic API to TEE-specific
+implementation.
+
+    With this option, application enclaves still call TEE-agnostic OE SDK API
+    `oe_get_attester_plugins()`.
+    But the OE SDK provides a definition of this function as a macro calling
+    into the TEE- and library-specific initialization functions of linked
+    enclave-side plugin libraries.
+
+    This option has the potential to support linking with multiple
+    enclave-side plugin libraries in a single application enclave build
+    configuration.
+    The OE SDK build environment is required to provide a proper
+    `oe_get_attester_plugins()` macro definition for every configuration
+    it supports.
+
+3. Every enclave-side plugin library registers its initialization function
+with the OE SDK framework.
+
+    The design document
+    [OpenEnclave Modules](https://github.com/jhand2/openenclave/blob/global_module_initialization/docs/DesignDocs/openenclave_modules.md)
+    describes how an enclave-side module can register its initialization
+    function for invocation on first entry to the enclave (upon enclave load).
+
+    With this option, the implementation of the OE SDK attestation framework
+    provides a hook for enclave-side plugin libraries to register their
+    initialization functions upon enclave load.
+    To support ordering in the initialization of multiple plugin libraries,
```

Comment:
Is ordering actually a requirement, or is this not needed?   (I ask because it wasn't mentioned anywhere earlier in the document, so if it is a requirement, please add a discussion of it earlier in the doc.)

Reply:
See @yentsanglee comment on ordering. This document does not try to decide whether that's a requirement or not. It only discusses a possible solution if ordering is needed.